nemo_skills/pipeline/nemo_evaluator.py

```diff
@@ -0,0 +1,856 @@
+    server_nodes: int = typer.Option(1, help="Nodes to allocate for the self-hosted server"),
+    server_port: Optional[int] = typer.Option(None, help="Port for the server; if unset uses free/random"),
+    server_args: Optional[str] = typer.Option(None, help="Extra args for server (passed through)"),
+    server_entrypoint: Optional[str] = typer.Option(None, help="Custom entrypoint for server (advanced)"),
+    server_container: Optional[str] = typer.Option(
+        None, help="Container key/image for server (defaults to cluster 'nemo-skills' if None)"
+    ),
+    server_base_url: Optional[str] = typer.Option(
+        None, help="Use an externally hosted server instead of self-hosting (e.g., http://host:port)"
+    ),
+    server_api_path: str = typer.Option("/v1/chat/completions", help="API path used for evaluator target url"),
+    server_health_path: str = typer.Option("/health", help="Health path used to wait for server readiness"),
+    # Optional judge self-hosted server (similar to main server)
+    judge_server_type: Optional[str] = typer.Option(
+        None,
+        help=("If set, self-host a judge server and co-schedule with evaluator. Supported values: vllm (preferred)."),
+    ),
+    judge_server_model: Optional[str] = typer.Option(
+        None, help="Model path/name to serve for judge (e.g., Qwen/Qwen3-32B-Instruct)"
+    ),
+    judge_server_gpus: int = typer.Option(0, help="GPUs to allocate for the judge server (0 = no judge server)"),
+    judge_server_nodes: int = typer.Option(1, help="Nodes to allocate for the judge server"),
+    judge_server_port: Optional[int] = typer.Option(None, help="Port for the judge server; if unset uses free/random"),
+    judge_server_args: Optional[str] = typer.Option(None, help="Extra args for judge server (passed through)"),
+    judge_server_entrypoint: Optional[str] = typer.Option(None, help="Custom entrypoint for judge server (advanced)"),
+    judge_server_container: Optional[str] = typer.Option(
+        None, help="Container key/image for judge server (defaults to cluster 'vllm' or 'nemo-skills')"
+    ),
+    judge_server_base_url: Optional[str] = typer.Option(
+        None, help="Use an externally hosted judge server instead of self-hosting (e.g., http://host:port)"
+    ),
+    judge_server_api_path: str = typer.Option("/v1/chat/completions", help="API path used for judge target url"),
+    judge_server_health_path: str = typer.Option(
+        "/health", help="Health path used to wait for judge server readiness"
+    ),
+    # Generic launcher override mechanisms
+    launcher_overlay: Optional[str] = typer.Option(
+        None,
+        help=(
+            "Overlay for the evaluator launcher Hydra config; accepts a path to YAML/JSON "
+            "or an inline YAML/JSON string. Flattened into ++key=value overrides unless explicitly set in ctx.args."
+        ),
+    ),
+    launcher_override: List[str] = typer.Option(
```

Comment:
what's the use-case of this and above? Maybe share an example to help understand